userspace/libsinsp/threadinfo.cpp

```diff
@@ -515,8 +515,22 @@ void sinsp_threadinfo::set_args(const char* args, size_t len)
 
 void sinsp_threadinfo::set_env(const char* env, size_t len)
 {
-	m_env.clear();
+	if (len == SCAP_MAX_ENV_SIZE && m_inspector->is_live())
+	{
+		// the environment is possibly truncated, try to read from /proc
+		// this may fail for short-lived processes
+		if (set_env_from_proc())
+		{
+#ifdef _DEBUG
```

Comment:
why do you need the define guards? isn't it enough to just log at SEV_DEBUG?

Reply:
I think I was worried about the overhead when log severity < SEV_DEBUG and the load is high. We can drop the guards if you prefer.

Reply:
`#ifdef` removed.